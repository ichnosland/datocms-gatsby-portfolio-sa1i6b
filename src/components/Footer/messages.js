/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */
import { defineMessages } from 'react-intl';


export default defineMessages({
  piva: {
    id: 'app.components.Footer.piva',
    defaultMessage: 'P. IVA 10623350013',
  },
});
