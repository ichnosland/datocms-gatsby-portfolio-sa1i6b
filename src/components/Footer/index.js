/**
*
* Footer
*
*/

import React from 'react';
//import { APPLICATION } from 'configuration';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

import ClearLink from '../ClearLink';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import messages from './messages';

const year = new Date().getFullYear();

export const FooterBlock = styled.footer`
position: relative;
width: 100%;
background-color: ${() => {
     return colore.landing.green;
  }};;
`;

export const Copy = styled.p`
  color: #fff;
  font-size: 1rem;
  text-align: center;
  padding: 20px 0;
  margin: 0 auto;
  a {
    display: inline;
    color: inherit;
    text-decoration: underline;
  }
  ${media.lt667`
    font-size: 15px;
  `}
  ${media.lt480`
    font-size: 12px;
  `}
`;
class Footer extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <FooterBlock>
        <Copy>© {year} <a href="https://maieuticallabs.it/" target="_blank">Maieutical Labs</a> • <ClearLink to="/contatti">Contatti</ClearLink> • <FormattedMessage {...messages.piva} /><span className="mediumUp"> • Tutti i diritti riservati</span>
        </Copy>
      </FooterBlock>
    );
  }
}

export default Footer;
