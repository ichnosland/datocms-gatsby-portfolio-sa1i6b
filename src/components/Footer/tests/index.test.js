import React from 'react';
import { shallow } from 'enzyme';

import Footer, { Copy, FooterBlock } from '../index';

describe('<Footer />', () => {
  it('should render a FooterBlock', () => {
    const renderedComponent = shallow(
      <Footer />
    );
    expect(renderedComponent.find(FooterBlock).length).toEqual(1);
  });
  it('should render a Copy', () => {
    const renderedComponent = shallow(
      <Footer />
    );
    expect(renderedComponent.find(Copy).length).toEqual(1);
  });
});

describe('<Copy />', () => {
  it('should render an <p> tag', () => {
    const renderedComponent = shallow(<Copy />);
    expect(renderedComponent.type()).toEqual('p');
  });
  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Copy id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });
});

describe('<FooterBlock />', () => {
  const OLD_ENV = process.env;
  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
    delete process.env.NODE_ENV;
  });
  it('should render an <footer> tag', () => {
    const renderedComponent = shallow(<FooterBlock />);
    expect(renderedComponent.type()).toEqual('footer');
  });
  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<FooterBlock id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });
  it('should render a green bg if APPLICATION === LandingAlatin', () => {
    process.env.APPLICATION = 'LandingAlatin';
    const FooterBlockComp = require('../index').FooterBlock; // eslint-disable-line global-require
    const renderedComponent = shallow(<FooterBlockComp />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render a red bg if APPLICATION === LandingItaca', () => {
    process.env.APPLICATION = 'LandingItaca';
    const FooterBlockComp = require('../index').FooterBlock; // eslint-disable-line global-require
    const renderedComponent = shallow(<FooterBlockComp />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
