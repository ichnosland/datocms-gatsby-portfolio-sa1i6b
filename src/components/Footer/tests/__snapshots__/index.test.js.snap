// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterBlock /> should render a green bg if APPLICATION === LandingAlatin 1`] = `
<footer
  className="hevhiW"
/>
`;

exports[`<FooterBlock /> should render a red bg if APPLICATION === LandingItaca 1`] = `
<footer
  className="iXbYXs"
/>
`;
